src/System.Net.HttpListener/src/System/Net/HttpListenerRequest.cs

```diff
@@ -86,6 +86,21 @@ public Encoding ContentEncoding
 
         private string RequestScheme => IsSecureConnection ? "https" : "http";
 
+        public Uri UrlReferrer
+        {
+            get
+            {
+                string referrer = Headers["referer"];
```

Comment:
HttpRequestHeader.Referer instead of "referer", same for other strings used in Headers in the PR..